kedro/io/data_catalog.py

```diff
@@ -328,31 +336,30 @@ class to be loaded is specified with the key ``type`` and their
     def _get_dataset(
         self, data_set_name: str, version: Version = None, suggest: bool = True
     ) -> AbstractDataSet:
-        if data_set_name not in self._data_sets:
-            # When a dataset is "used" in the pipeline that's not in the recorded catalog datasets,
-            # try to match it against the data factories in the catalog. If it's a match,
-            # resolve it to a dataset instance and add it to the catalog, so it only needs
-            # to be matched once and not everytime the dataset is used in the pipeline.
-            if self.exists_in_catalog_config(data_set_name):
+        if self.exists_in_catalog_config(data_set_name):
+            if data_set_name not in self._data_sets:
+                # When a dataset is "used" in the pipeline that's not in the
+                # recorded catalog datasets, match it against the data factories
+                # in the catalog, resolve it to a dataset instance and add it to
+                # the catalog, so it only needs to be matched once and not
+                # everytime the dataset is used in the pipeline.
                 pattern = self._pattern_matches_cache[data_set_name]
                 matched_dataset = self._resolve_dataset(data_set_name, pattern)
                 self.add(data_set_name, matched_dataset)
-            else:
-                error_msg = f"Dataset '{data_set_name}' not found in the catalog"
-
-                # Flag to turn on/off fuzzy-matching which can be time consuming and
-                # slow down plugins like `kedro-viz`
-                if suggest:
-                    matches = difflib.get_close_matches(
-                        data_set_name, self._data_sets.keys()
-                    )
-                    if matches:
-                        suggestions = ", ".join(matches)
-                        error_msg += (
-                            f" - did you mean one of these instead: {suggestions}"
-                        )
-
-                raise DatasetNotFoundError(error_msg)
+        else:
+            error_msg = f"Dataset '{data_set_name}' not found in the catalog"
+
+            # Flag to turn on/off fuzzy-matching which can be time consuming and
+            # slow down plugins like `kedro-viz`
+            if suggest:
+                matches = difflib.get_close_matches(
+                    data_set_name, self._data_sets.keys()
+                )
+                if matches:
+                    suggestions = ", ".join(matches)
+                    error_msg += f" - did you mean one of these instead: {suggestions}"
+
+            raise DatasetNotFoundError(error_msg)
```

Comment:
It's small change but I think this is easier to understand now and the nested logic is less complex.